spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1139,6 +1139,26 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  // TODO: test currently fails with
+  // Unsupported Data Type: Spark `expm1` function expects Float64, got Utf8
+  test("expm1") {
+    // TODO test with null inputs
+    val testValues = Seq(
+      -1,
+      0,
+      +1,
+      Double.MinValue,
+      Double.MaxValue,
+      Double.NaN,
+      Double.MinPositiveValue,
+      Double.PositiveInfinity,
+      Double.NegativeInfinity)
+    val testValuesRepeated = testValues.flatMap(v => Seq.fill(1000)(v))
+    withParquetTable(testValuesRepeated.map(n => (n, n)), "tbl") {
+      checkSparkAnswerWithTol("SELECT expm1(_1) FROM tbl")
```

Comment:
This test fails with `Unsupported Data Type: Spark `expm1` function expects Float64, got Utf8`, but if I change the test to call `tan` instead of `expm1`, then it passes. I do not understand where `Utf8` is coming from yet.